pkg/server/endpoints/authorized_entryfetcher.go

```diff
@@ -62,12 +62,14 @@ func (a *AuthorizedEntryFetcherWithEventsBasedCache) RunUpdateCacheTask(ctx cont
 		select {
 		case <-ctx.Done():
 			a.log.Debug("Stopping in-memory entry cache hydrator")
-			return nil
+			return ctx.Err()
 		case <-a.clk.After(a.cacheReloadInterval):
-			err := a.updateCache(ctx)
-			if err != nil {
+			if err := a.updateCache(ctx); err != nil {
 				a.log.WithError(err).Error("Failed to update entry cache")
 			}
+			if pruned := a.cache.PruneExpiredAgents(); pruned > 0 {
+				a.log.Debugf("Pruned %d expired agents from entry cache", pruned)
```

Comment:
Should use fields rather than formatted strings:
```suggestion
				a.log.WithField("count", pruned).Debug("Pruned expired agents from entry cache")
```